core/src/com/unciv/logic/city/CityConstructions.kt

```diff
@@ -323,6 +327,23 @@ class CityConstructions {
         }
     }
 
+    private fun constructionBegun(construction: IConstruction) {
+        if (construction is Building) {
```

Comment:
"Exit quickly" is a Good Practice (tm) since it both reduces doubt and indentation and makes functions easier to parse

if (construction !is Building) return
if (construction.uniqueObjects.none { it.placeholderText == "Triggers a global alert upon build start" }) return